Mathlib/Data/List/Count.lean

```diff
@@ -13,16 +13,34 @@ This file proves basic properties of `List.countP` and `List.count`, which count
 elements of a list satisfying a predicate and equal to a given element respectively.
 -/
 
+assert_not_exists Monoid
 assert_not_exists Set.range
-assert_not_exists GroupWithZero
-assert_not_exists Ring
 
 open Nat
 
 variable {α : Type*}
 
 namespace List
 
+lemma countP_erase [DecidableEq α] (p : α → Bool) (l : List α) (a : α) :
+    countP p (l.erase a) = countP p l - if a ∈ l ∧ p a then 1 else 0 := by
+  rw [countP_eq_length_filter, countP_eq_length_filter, ← erase_filter, length_erase]
+  aesop
+
+lemma count_diff [DecidableEq α] (a : α) (l₁ : List α) :
+    ∀ l₂, count a (l₁.diff l₂) = count a l₁ - count a l₂
+  | [] => rfl
+  | b :: l₂ => by
+    simp only [diff_cons, count_diff, count_erase, beq_iff_eq, Nat.sub_right_comm, count_cons,
+      Nat.sub_add_eq]
+
+lemma countP_diff [DecidableEq α] {l₁ l₂ : List α} (hl : l₂ <+~ l₁) (p : α → Bool) :
+    countP p (l₁.diff l₂) = countP p l₁ - countP p l₂ := by
+  refine (Nat.sub_eq_of_eq_add ?_).symm
+  rw [← countP_append]
+  exact ((subperm_append_diff_self_of_count_le <| subperm_ext_iff.1 hl).symm.trans
+    perm_append_comm).countP_eq _
+
 /-! ### count -/
 
 section Count
```

Comment:
I think we should either put your new results in this section, or preferably delete the section outright. What do you think?

Reply:
You're right. It's a silly section